test/integration/integration.cc

```diff
@@ -282,6 +282,14 @@ BaseIntegrationTest::BaseIntegrationTest(Network::Address::IpVersion version,
           },
           version, config) {}
 
+BaseIntegrationTest::~BaseIntegrationTest() {
+  // Tear down the fake upstream before the test server.
+  // When the HTTP codecs do runtime checks, it is important to finish all
+  // runtime access before the server, and the runtime singleton, go away.
+  fake_upstreams_.clear();
```

Comment:
I think you could also reverse declarations of `test_server_` and `fake_upstreams_` in the class and have C++ enforce the cleanup order. If you are into that sort of thing.

Reply:
Yeah, I thought about that, but I was concerned there might be other things that needed to go before the server.

I should just be non-lazy and put it at the top :-P

Reply:
Actually yeah, if I move it up something else fails.
How about we land this to remove the flake and I file a tracking issue for cleaning up the tests a bit?

Reply:
https://github.com/envoyproxy/envoy/issues/11553